src/Workspaces/Core/Portable/Workspace/Solution/ProjectOrDocumentId.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal readonly struct ProjectOrDocumentId : IEquatable<ProjectOrDocumentId>
```

Comment:
Should this type be kept in some solution crawler/work coordinator code until we have a reason to keep it in the more general folder? I admit I can't imagine off the top of my head a use outside that code -- but in that code it's obviously useful.